frontend/app/routes/protected/person-case/secondary-doc.tsx

```diff
@@ -0,0 +1,179 @@
+import { useId } from 'react';
+
+import { data, useFetcher } from 'react-router';
+import type { RouteHandle } from 'react-router';
+
+import { faExclamationCircle, faXmark } from '@fortawesome/free-solid-svg-icons';
+import type { SessionData } from 'express-session';
+import { useTranslation } from 'react-i18next';
+import * as v from 'valibot';
+
+import type { Route, Info } from './+types/secondary-doc';
+
+import { requireAuth } from '~/.server/utils/auth-utils';
+import { i18nRedirect } from '~/.server/utils/route-utils';
+import { Button } from '~/components/button';
+import { DatePickerField } from '~/components/date-picker-field';
+import { FetcherErrorSummary } from '~/components/error-summary';
+import { InputFile } from '~/components/input-file';
+import { InputRadios } from '~/components/input-radios';
+import { PageTitle } from '~/components/page-title';
+import { Progress } from '~/components/progress';
+import { AppError } from '~/errors/app-error';
+import { ErrorCodes } from '~/errors/error-codes';
+import { getFixedT } from '~/i18n-config.server';
+import { handle as parentHandle } from '~/routes/protected/layout';
+import { getLanguage } from '~/utils/i18n-utils';
+
+type SecondaryDocumentsSessionData = NonNullable<SessionData['inPersonSINCase']['secondaryDocument']>;
+
+const VALID_DOCTYPE = ['passport', 'id-card', 'other-id'];
+
+export const handle = {
+  i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
+} as const satisfies RouteHandle;
+
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  return {
+    documentTitle: t('protected:secondary-identity-document.page-title'),
+    defaultFormValues: context.session.inPersonSINCase?.secondaryDocument,
+  };
+}
+
+export function meta({ data }: Route.MetaArgs) {
+  return [{ title: data.documentTitle }];
+}
+
+export async function action({ context, request }: Route.ActionArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const lang = getLanguage(request);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  const formData = await request.formData();
+  const action = formData.get('action');
+  const maxImageSize = 1024 * 1024 * 15; //Max image size is 15 MB
```

Comment:
Nitpicky suggestion: add the units to this so it's clear.

`const maxImageSizeBits`